packages/svelte/src/internal/client/runtime.js

```diff
@@ -582,17 +583,21 @@ export function schedule_effect(signal) {
 
 	var effect = signal;
 
-	while (effect.parent !== null) {
-		effect = effect.parent;
-		var flags = effect.f;
+	while (true) {
+		if ((effect.f & EFFECT_HAS_DIRTY_CHILDREN) !== 0) {
+			return;
+		}
+
+		var parent = effect.parent;
 
-		if ((flags & BRANCH_EFFECT) !== 0) {
-			if ((flags & CLEAN) === 0) return;
-			set_signal_status(effect, MAYBE_DIRTY);
+		if (parent === null) {
+			queued_root_effects.push(effect);
+			return;
 		}
-	}
 
-	queued_root_effects.push(effect);
+		effect.f |= EFFECT_HAS_DIRTY_CHILDREN;
```

Comment:
This line is also super expensive, it's now 12ms, before the:

```js
set_signal_status(effect, MAYBE_DIRTY);
```

was only 1.9ms